index.js

```diff
@@ -391,6 +393,12 @@ function wrapPromise() {
     return function wrappedThen() {
       var promise = this;
       var next = original.apply(promise, Array.prototype.map.call(arguments, bind));
+
+      next.__asl_wrapper = function proxyWrapper(ctx, fn, val, last) {
+        promise.__asl_wrapper(ctx, function () {}, null, next)
+        return next.__asl_wrapper(ctx, fn, val, last)
```

Comment:
@Qard how do you feel about this? feels really dirty to have next.__asl_wrapper call a function that will replace the __asl_wrapper function on next, and then invoke that replaced function.  I couldn't think of a clean way around this, but open to suggestions